src/frontend/ui/tool/mod.rs

```diff
@@ -7,17 +7,18 @@ use super::drawable::Drawable;
 
 pub mod crop;
 
-#[derive(Debug, PartialEq, Eq, Hash)]
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
 pub enum ToolIdentifier {
     Crop,
 }
 
+#[derive(Debug)]
 pub enum ToolCommand {
     Noop,
     Crop(Rectangle),
 }
 
-pub trait Tool {
+pub trait Tool: std::fmt::Debug {
```

Comment:
```suggestion
pub trait Tool : std::fmt::Debug {
```

Reply:
Can't be done. Formatter says:
![Uploading peepoNo.png…]()